sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/Signer.cs

```diff
@@ -143,7 +155,19 @@ private static async Task SignRequestAsync(IRequestContext requestContext)
 
             // credentials would be null in the case of anonymous users getting public resources from S3
             if (immutableCredentials == null && requestContext.Signer.RequiresCredentials)
+            {
                 return;
+            }
+
+            // The signer interface expects immutable credentials, which does not match the AWSToken structure.
+            // We'll manually set a property so that the bearer token still works as expected.
+            if (requestContext.Signer is BearerTokenSigner tokenSigner)
+            {
+                if (requestContext.Identity is AWSToken resolvedToken)
+                {
+                    tokenSigner.ResolvedToken = resolvedToken.Token;
```

Comment:
We are mutating state on the bearer token signer which is being use across threads. We need a safer way to do this.